### ISSUE TITLE
Ability to prepare, but not enable, a synced folder

### ISSUE THREAD

**Issue by jsternberg** [mitchellh/vagrant#6835]
I want to make a Vagrantfile where the provisioner runs before the synced folder is mounted. This is similar to #936 and #2701. In those issues, some workarounds were mentioned, but I haven't managed to get them to work for myself.

I need to have some setup run before the folder is mounted because I need the synced folder to be inside another folder that doesn't exist in the Virtual Machine until after my provisioning. I either need the option for synced folders to be done after provisioning or I could also use preparing the synced folder and disable the auto mount so I can use my provisioner to call `mount` manually.

I tried to use `disabled: true` but this seems to cause the folder to never be prepared. I tried some of the advice mentioned [here](https://github.com/mitchellh/vagrant/issues/936#issuecomment-23467018), but the following don't work:

```
# requires at least two arguments
config.vm.synced_folder "."
# the guestpath is used as the name and the guestpath, causes an error when booting
config.vm.synced_folder ".", nil
# sets the name correctly, but errors out because the guestpath gets used at some point
options = config.vm.synced_folder ".", "name"
options[:guestpath] = nil
```

Is there anyway to do what I want?

I am using Vagrant 1.8.1.

**Comment by sethvargo** [comment]
Hi @jsternberg 

Thank you for opening an issue. This is a duplicate of https://github.com/mitchellh/vagrant/issues/936 and unfortunately there is not very much Vagrant can do in terms of remounting shared folders. The hypervisior needs to be told about the synced folders _before_ the machine boots, so it is not currently possible to alter that for most hypervisors. If that ever changes, we can definitely revisit this :smile:

**Comment by jsternberg** [comment]
@sethvargo I read #936 and this one is slightly different. I want the ability to be able to add the synced folder, but not to perform the mount. The pull request I opened at #6836 doesn't do this ticket exactly as stated, but it fulfills the primary need which is to have a method to skip the mounting step for VirtualBox Shared Folders.

**Comment by jsternberg** [comment]
@sethvargo is it possible to look at my PR? The request isn't to modify how Vagrant mounts shared folders. I just want Vagrant to add the shared folder and not perform the mount. I think the PR I created in #6836 does this in a way that's consistent with how it was suggested in the issues I linked to.

The PR gives a lot more flexibility so it is possible to mount the drive to locations created by the provisioner script.

Thanks.
